### ISSUE TITLE
function _installer_browser_is_project_enabled always returns false for layouts and always true for themes.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#2336]
https://github.com/backdrop/backdrop/blob/1.x/core/modules/installer/installer.browser.inc#L18

Discovered that while putting together a PR for #2335 

PS: this is similar/related to #1709

---
~~PR by @klonos: https://github.com/backdrop/backdrop/pull/1644~~
PR by @klonos and @quicksketch: https://github.com/backdrop/backdrop/pull/1728

**Comment by klonos** [comment]
...while at it, I also removed a stray `dpm` from layout.admin.inc 😉

**Comment by klonos** [comment]
...turns out the logic in the same function for the themes is not correct either.

**Comment by klonos** [comment]
The PR passes for php70, but fails for php53. Not sure if the failures are random and would love some feedback as to what I may be doing wrong.

**Comment by quicksketch** [comment]
> The PR passes for php70, but fails for php53.

Yeah, there's an issue with the use of the `empty()` function, which works slightly differently on newer versions of PHP. See https://github.com/backdrop/backdrop/pull/1644#pullrequestreview-15633495

I filed a PR fixing this issue at https://github.com/backdrop/backdrop/pull/1728

Thanks for finding that stray dpm() as well!

**Comment by quicksketch** [comment]
I've merged https://github.com/backdrop/backdrop/pull/1728 into 1.x and 1.5.x.

**Comment by klonos** [comment]
Actually, this still does not work for layouts...

**Comment by klonos** [comment]
...to reproduce the issue:

- install a fresh 1.6.0.
- head to the "Install layouts" page (`admin/structure/layouts/install`).
- add *Senjo* to the installation queue and install it.
- after installation, go to the "Install layouts" page again -> *Senjo* is correctly shown as "Already enabled"
- add *2 Columns Flex* to the installation queue and install it.
- after installation, go to the "Install layouts" page again -> *Senjo* is still correctly shown as "Already enabled" but *2 Columns Flex* still has the "Add to Installation queue" link available.

With the commit in this issue, [this code has changed](https://github.com/backdrop/backdrop/blob/1.x/core/modules/installer/installer.browser.inc#L27):

```php
    case 'layout':
      $excluded_templates = config_get('layout.settings', 'excluded_templates');
      return layout_get_layout_template_info($name) && !in_array($name, $excluded_templates);
      break;
```

...the thing is that the `$name` variable passed to the `_installer_browser_is_project_enabled` function is a layout *project* name while `layout_get_layout_template_info()` returns the layout *template* names (thus `layout_get_layout_template_info($name)` will always return `FALSE`). Same goes for `$excluded_templates` which also returns an array of layout *template* names. So the `$name` variable for layouts that include more than one templates (and thus template name != project name) will never be in the `$excluded_templates` array.

On top of that, since we are using `layout_get_layout_template_info()`, we need to first fix #984 and then pass the `$rebuild` argument to it inside this function.

**Comment by jenlampton** [comment]
Pushing this to the 1.6.2 milestone since the PR needs work and we are releasing 1.6.1 tonight.

**Comment by docwilmot** [comment]
Can we get this finished @klonos ? Seems like a simple cleanup, and has been sitting for way too long.

**Comment by BWPanda** [comment]
Since it's been a few years since any work was done here, I'm removing the milestone and adding back the candidate label. Once there's an RTBC PR, we can add the milestone back.

**Comment by jenlampton** [comment]
Can you check the criteria for bug-fix milestones? IIRC the PR does not
need to be RTBC.

**Comment by BWPanda** [comment]
OK, maybe not exactly RTBC, but at least in a near-ready state. To me, two years with 'needs work' is not near-ready... I just think its ridiculous to have people needing to change the milestone every release for an issue that's clearly not a priority for anyone currently...

**Comment by jenlampton** [comment]
> I just think its ridiculous to have people needing to change the milestone every release for an issue that's clearly not a priority for anyone currently...

It's no trouble, there's a single bulk-aciton to move all issues at once.

**Comment by BWPanda** [comment]
@jenlampton Sure, but it also adds unnecessary noise to the issue. I had to "Show hidden items" twice to see everything in this issue.

Also, removing PR labels since there's no PR (AFAIK) for the changes since the last PR was merged...
